bedrock/bedrock-v388/src/main/java/com/nukkitx/protocol/bedrock/v388/serializer/AvailableCommandsSerializer_v388.java

```diff
@@ -158,7 +158,9 @@ public void deserialize(ByteBuf buffer, BedrockPacketHelper helper, AvailableCom
                     flagList, command.getPermission(), aliases, overloads));
         }
 
-        // Constraints
-        helper.readArray(buffer, packet.getConstraints(), buf -> helper.readCommandEnumConstraints(buf, enums, enumValues));
+        // Read constraints if included
+        if (buffer.readableBytes() > 1){
```

Comment:
What were you testing on to get this because I don't think it's vanilla behaviour. I would like to avoid putting in checks like this if possible

Reply:
I have already asked you for better solution on Discord. Let me summarize it here.
For some reason reading it may throw index out of bounds exception. I think it is in case when array is empty.
Looks like servers before 1.16 are not sending that 0 byte at the end. Therefore I have added check for readable bytes.

Reply:
Yes, you mentioned this on Discord and I just had a look through the vanilla packet serialisation which always encodes the length of the constraints at the end. If you received a packet without the constraint length, could you give the server version and software?

Reply:
It happens with Nukkit before 1.16 update.

Reply:
Just checked it again. I refferenced to commit of one of the unofficial nukkit forks. I guess I can remove the check than.